types/gov.go

```diff
@@ -12,20 +13,55 @@ const (
 	ProposalStatusInvalid = "PROPOSAL_STATUS_INVALID"
 )
 
-// GovParams contains the data of the x/gov module parameters
+type DepositParams struct {
+	MinDeposit       sdk.Coins `json:"min_deposit,omitempty" yaml:"min_deposit"`
+	MaxDepositPeriod string    `json:"max_deposit_period,omitempty" yaml:"max_deposit_period"`
+}
+
+func NewdepositParam(d govtypes.DepositParams) DepositParams {
+	return DepositParams{
+		MinDeposit:       d.MinDeposit,
+		MaxDepositPeriod: strconv.FormatInt(d.MaxDepositPeriod.Nanoseconds(), 10) + "n",
+	}
+}
+
+type VotingParams struct {
+	VotingPeriod string `json:"voting_period,omitempty" yaml:"voting_period"`
+}
+
+func NewvotingParams(v govtypes.VotingParams) VotingParams {
+	return VotingParams{
+		VotingPeriod: strconv.FormatInt(v.VotingPeriod.Nanoseconds(), 10) + "n",
+	}
+}
+
 type GovParams struct {
-	govtypes.Params
-	Height int64
+	DepositParams DepositParams        `json:"deposit_params" yaml:"deposit_params"`
+	VotingParams  VotingParams         `json:"voting_params" yaml:"voting_params"`
+	TallyParams   govtypes.TallyParams `json:"tally_params" yaml:"tally_params"`
+	Height        int64                `json:"height" ymal:"height"`
 }
 
-// NewGovParams allows to build a new GovParams instance
-func NewGovParams(params govtypes.Params, height int64) *GovParams {
+func NewGovParams(v VotingParams, d DepositParams, t govtypes.TallyParams, height int64) *GovParams {
 	return &GovParams{
-		Params: params,
-		Height: height,
+		DepositParams: d,
+		VotingParams:  v,
+		TallyParams:   t,
+		Height:        height,
 	}
 }
 
+func (*VotingParams) ProtoMessage()  {}
+func (*DepositParams) ProtoMessage() {}
+func (m *VotingParams) Reset()       { *m = VotingParams{} }
+func (m *DepositParams) Reset()      { *m = DepositParams{} }
+func (m VotingParams) String() string {
+	return m.VotingPeriod
+}
+func (m DepositParams) String() string {
+	return m.MaxDepositPeriod
+}
+
```

Comment:
What is this used for? Can we delete this?

Reply:
Deleted